contrib/mesos/docs/scheduler.md

```diff
@@ -0,0 +1,115 @@
+# Kubernetes-Mesos Scheduler
+
+## Labels and Mesos Agent Attributes
+
+The scheduler of Kubernetes on Mesos takes [labels][1] into account: it matches
+specified labels in pod specs with defined labels of nodes.
+
+In addition to user defined labels, [attributes of Mesos agents][2] are converted
+into node labels by the scheduler, following the pattern
+
+```yaml
+k8s.mesosphere.io/attribute-<name>: value
+```
+
+As an example, an attribute `generation: 2015` will result in the node label
+
+```yaml
+k8s.mesosphere.io/attribute-generation: 2015
+```
+
+and be used to schedule pods onto node which are of generation 2015.
+
+**Note:** Node labels which originate from Mesos agent attributes must not be
+modified by the user using e.g. kubectl. Instead the Kubernetes executor will
+notice when a mesos-slave is restarted with new attributes and will modify the
+node object accordingly on the apiserver.
+
+## Resource Roles
+
+A Mesos cluster can be statically partitioned using [resources roles][2]. Each
+role is assigned such a role (`*` is the default role, if none is explicitly
+assigned in the mesos-slave command line) and the Mesos master will send offers to
+frameworks for `*` resources and – optionally – for one extra role that a
+framework is assigned to. Right now only one such extra role for a framework is
+supported.
+
+### Configuring Roles for the Scheduler
+
+Every Mesos framework scheduler can choose among the offered `*` resources and
+those of the extra role. The Kubernetes-Mesos scheduler supports this by setting
+the framework roles in the scheduler command line, e.g.
+
+```bash
+$ km scheduler ... --mesos-roles=*,role1 ...
+```
+
+This will tell the Kubernetes-Mesos scheduler to default to using `*` resources
+if a pod is not specially assigned to another role. Moreover, the extra role
+`role1` is allowed, i.e. the Mesos master will send resources or role `role1`
+to the Kubernetes scheduler.
+
+Note the following restrictions and possibilities:
+- Due to the restrictions of Mesos, only one extra role may be provided on the
+  command line.
+- It is allowed to only pass an extra role without the `*`, e.g. `--mesos-roles=role1`.
+  This means that no `*` resources should be considered by the scheduler at all.
+- It is allowed to pass the extra role first, e.g. `--mesos-roles=role1,*`.
+  This means that `role1` is the default role for pods without special role
+  assignment (see below). But `*` resources would be considered for pods with a special `*`
+  assignment.
+
+### Specifying Roles for Pods
+
+By default a pod is scheduled using resources of the role which comes first in
```

Comment:
By default no pods are scheduled onto the public slaves, only when you explicitly say so.

When Mesos support more than one role for a framework, we can extend this to a subset of roles which are default, and some more which are not.